src/sagemaker/jumpstart/utils.py

```diff
@@ -109,7 +109,8 @@ def get_jumpstart_gated_content_bucket(
     accessors.JumpStartModelsAccessor.set_jumpstart_gated_content_bucket(gated_bucket_to_return)
 
     if gated_bucket_to_return != old_gated_content_bucket:
-        accessors.JumpStartModelsAccessor.reset_cache()
+        if old_gated_content_bucket is not None:
```

Comment:
nit: This seems to be copied from [get_jumpstart_content_bucket()](https://github.com/aws/sagemaker-python-sdk/blob/ef8dd31d6b5e915c746788e9ec3cdb308f227dd2/src/sagemaker/jumpstart/utils.py#L119). Can we pass in these bucket names in and merge these two functions?

Reply:
for now, let's keep as is. i agree, there is code duplication that should be removed.

btw, this change is for ensuring the wildcard logging happens a single time. basically, what is happening is that when the model is initialized, the cache got reset, causing the wildcard warning to get emitted twice.

Reply:
Ah I see, that makes a lot of sense. Nice catch!